Formula/open-sp.rb

```diff
@@ -12,14 +12,37 @@ class OpenSp < Formula
     sha256 "d7b79be390f3c2b2a823e1156d896200db397dffb6cb6e6712d27539e05ca18b" => :yosemite
   end
 
+  depends_on "docbook" => :build
+  depends_on "xmlto" => :build
+  depends_on "ghostscript" => :build
+
   def install
     system "./configure", "--disable-debug", "--disable-dependency-tracking",
                           "--prefix=#{prefix}",
                           "--mandir=#{man}",
-                          "--disable-doc-build",
                           "--enable-http",
-                          "--enable-default-catalog=#{etc}/sgml/catalog",
-                          "--enable-default-search-path=#{HOMEBREW_PREFIX}/share/sgml"
+                          "--enable-default-catalog=#{etc}/sgml/catalog"
+
+    ENV["XML_CATALOG_FILES"] = "#{etc}/xml/catalog"
     system "make", "pkgdatadir=#{share}/sgml/opensp", "install"
   end
+
+  test do
+    sgml = <<-EOF
+<!DOCTYPE TESTDOC [
+
+<!ELEMENT TESTDOC - - (TESTELEMENT)+>
+<!ELEMENT TESTELEMENT - - (#PCDATA)>
+
+]>
+<TESTDOC>
+  <TESTELEMENT>Hello</TESTELEMENT>
+</TESTDOC>
+EOF
+    File.open('eg.sgml', 'w:utf-8') do |f|
+      f.write(sgml)
+    end
+
+    system "onsgmls", "--warning=type-valid", "eg.sgml"
```

Comment:
can we assert something here with assert_equal or assert_match or assert_predicate and not just check for 0 exit code?

Reply:
The zero-exit-code check here is actually a pretty comprehensive check - the invocation will exit non-zero unless the input SGML document can be parsed and is valid against the internal DTD.

Reply:
OK